lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -542,17 +732,48 @@ def get_sec_group_list(groups):
     # Build list of remote security groups
     remote_security_groups = []
     for group in groups:
-        remote_security_groups.append(group.id.encode())
+        remote_security_groups.append(group["GroupId"].encode())
 
     return remote_security_groups
 
 
 def _get_vpc_id(connection, module, subnet_id):
 
     try:
-        return connection.get_all_subnets(subnet_ids=[subnet_id])[0].vpc_id
-    except BotoServerError as e:
-        module.fail_json(msg=e.message)
+        subnets = connection.describe_subnets(SubnetIds=[subnet_id])
+        return subnets["Subnets"][0]["VpcId"]
+    except botocore.exceptions.ClientError as e:
+        module.fail_json_aws(e, "Failed to get vpc_id for {0}".format(subnet_id))
+
+
+@AWSRetry.jittered_backoff()
+def manage_tags(eni, new_tags, purge_tags, connection):
```

Comment:
This doesn't exist in module_utils?

Reply:
No.  I can move it there.  I found this method in ec2_instance, so I can move it to module_utils and refactor that occurance, too.

Reply:
The tag handling in this function is currently ENI specific.  It could be made more generic, but AWS is inconsistent on where it sets the tags.  For instances its in "Tags" and for ENI it is in "TagSet".  So I punted for now.